src/ui/components/Location/component.ts

```diff
@@ -169,33 +158,65 @@ export default class LocationComponent extends Component {
   }
 
   async loadFromAPI(locationSignature: RecordSignature) {
-    let { store } = this.args;
+    let { store, localStore } = this.args;
     try {
-      let location: Location = await store.query((q) =>
+      this.location = await store.query((q) =>
         q.findRecord(locationSignature)
       );
-      let measurements: Measurement[] = await store.query((q) =>
+    } catch {
+      // Fail silently
+    }
+    try {
+      let transform = await localStore.pull((q) =>
+        q.findRelatedRecords(locationSignature, 'measurements')
+      );
+
+      // Remove old data
+      let cachedResults = this.measurements = store.cache.query((q) =>
         q.findRelatedRecords(locationSignature, 'measurements')
       );
 
-      if (location && measurements) {
-        this.location = location;
-        this.measurements = measurements;
+      // Remove relationships
+      await store.update((t) => {
+        return cachedResults.map((result) => {
+          let measurementSignature = { type: 'measurement', id: result.id };
+          return t.removeFromRelatedRecords(
+            locationSignature,
+            'measurements',
+            measurementSignature
+          );
+        });
+      });
 
-        // Remember we saw this location
-        let currentDate = new Date().toISOString();
-        store.update((t) =>
-          t.replaceAttribute(locationSignature, 'visitedAt', currentDate)
-        );
-      }
+      // Remove old records
+      await store.update((t) => {
+        return cachedResults.map((result) => {
+          let measurementSignature = { type: 'measurement', id: result.id };
+          return t.removeRecord(
+            measurementSignature
+          );
+        });
+      });
+
+      // Add new data to store
+      await store.sync(transform);
```

Comment:
Will there be any data though? I don't see where the `remoteStore` is ever used and data is being loaded?